src/connectionManager.ts

```diff
@@ -0,0 +1,74 @@
+export interface Context {
+  id: string;
+  resourceID: string;
+  resourceType: string;
+  resourceDisplay?: string;
+}
+
+export interface Connection {
+  id: string;
+  url: URL;
+  context: Context[];
+}
+
+export class ConnectionManager {
+  private connections: Connection[];
+  private currentConnection?: Connection;
+
+  constructor() {
+    this.connections = [];
+  }
+
+  public getAllConnections(): Connection[] {
+    return this.connections;
+  }
+
+  public getCurrentConnection(): Connection | undefined {
+    return this.currentConnection;
+  }
+
+  public setCurrentConnection(connection: Connection | undefined) {
+    this.currentConnection = connection;
+  }
+
+  public addConnection(connection: Connection): void {
+    this.connections?.push(connection);
+  }
+
+  // TODO
+  public testConnection(connection: Connection): void {}
+
+  // TODO
+  public upsertConnection(connection: Connection): void {}
+
+  // TODO
+  public updateConnection(connection: Connection): void {}
```

Comment:
Wondering if we need both `update` and `upsert`? What's a case where we'd need `update` that wouldn't be covered by `upsert`? (Applies to `update/upsertContext` as well.)

Reply:
Good point. I wonder if we even want add either? Maybe just upsert?